lib/web/app/middleware.go

```diff
@@ -87,7 +87,12 @@ func (h *Handler) redirectToLauncher(w http.ResponseWriter, r *http.Request, p l
 		return trace.Wrap(err)
 	}
 
-	urlString := makeAppRedirectURL(r, h.c.WebPublicAddr, addr.Host(), p)
+	var proxyPublicAddrs []string
+	for _, proxyAddr := range h.c.ProxyPublicAddrs {
+		proxyPublicAddrs = append(proxyPublicAddrs, proxyAddr.String())
```

Comment:
nit: explain why `.String()` is used. 
```suggestion
	for _, proxyAddr := range h.c.ProxyPublicAddrs {
	        // preserving full `host:port` to support proxy hosted on non-standard https port.
		proxyPublicAddrs = append(proxyPublicAddrs, proxyAddr.String())
```